packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterItem.tsx

```diff
@@ -0,0 +1,298 @@
+              apiRef.current.showHeaderFilterMenu(colDef.field);
+              break;
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+            break;
+          case 'Tab': {
+            if (isEditing) {
+              const fieldToFocus = columnFields[colIndex + (event.shiftKey ? -1 : 1)] ?? null;
+
+              if (fieldToFocus) {
+                apiRef.current.startHeaderFilterEditMode(fieldToFocus);
+                apiRef.current.setColumnHeaderFilterFocus(fieldToFocus, event);
+              }
+            }
+            break;
+          }
+          default:
+            if (isEditing || event.metaKey || event.ctrlKey || event.altKey || event.shiftKey) {
+              break;
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+            break;
+        }
+      },
+      [apiRef, colDef.field, colIndex, columnFields, headerFilterMenuRef, isEditing, isMenuOpen],
+    );
+
+    const publish = React.useCallback(
+      (eventName: keyof GridHeaderFilterEventLookup, propHandler?: React.EventHandler<any>) =>
+        (event: React.SyntheticEvent) => {
+          apiRef.current.publishEvent(
+            eventName,
+            apiRef.current.getColumnHeaderParams(colDef.field),
+            event as any,
+          );
+          if (propHandler) {
+            propHandler(event);
+          }
+        },
+      [apiRef, colDef.field],
+    );
+
+    const onClick = React.useCallback(
+      (event: React.MouseEvent) => {
+        if (!hasFocus) {
+          apiRef.current.startHeaderFilterEditMode(colDef.field);
+          apiRef.current.setColumnHeaderFilterFocus(colDef.field, event);
+        }
+      },
+      [apiRef, colDef.field, hasFocus],
+    );
+
+    const mouseEventsHandlers = React.useMemo(
+      () => ({
+        onKeyDown: publish('headerFilterKeyDown', onKeyDown),
+        onClick: publish('headerFilterClick', onClick),
+      }),
+      [onClick, onKeyDown, publish],
+    );
+
+    const ownerState = {
+      ...rootProps,
+      colDef,
+    };
+
+    const classes = useUtilityClasses(ownerState as OwnerState);
+
+    const isNoInputOperator = NO_INPUT_OPERATORS[colDef.type!]?.includes(item.operator);
+    const isFilterActive = hasFocus || Boolean(item?.value) || isNoInputOperator;
+    const label =
+      OPERATOR_LABEL_MAPPING[item.operator] ??
+      apiRef.current.getLocaleText(
+        `filterOperator${capitalize(item.operator)}` as 'filterOperatorContains',
+      );
+
+    return (
+      <div
+        className={clsx(classes.root, headerClassName)}
+        ref={handleRef}
+        style={{
+          height,
+          width,
+          minWidth: width,
+          maxWidth: width,
+        }}
+        role="columnheader"
+        aria-colindex={colIndex + 1}
+        aria-label={headerFilterComponent == null ? colDef.headerName ?? colDef.field : undefined}
+        {...other}
+        {...mouseEventsHandlers}
```

Comment:
I created this sandbox with one column that is not filterable:
https://codesandbox.io/s/floral-wildflower-8obfkj?file=/demo.tsx

And after clicking on the empty header filter, the keyboard navigation doesn't work anymore (it enters editing mode)

https://user-images.githubusercontent.com/13808724/235479991-789e6703-b319-4f1b-87c1-89b91fb425a1.mov

Maybe we can return earlier and avoid that?

Nit: Another option would be offloading keyboard navigation to header filter components.
If we allow them to control whether navigation keys should move the focus to a different cell or not, then we might not even need the global state for this.

Reply:
Same happens when using `renderHeaderFilter`: https://codesandbox.io/s/delicate-thunder-yxjmcv?file=/demo.tsx

Reply:
Thanks for reporting the issue, the reason why it was happening like this is because we were deliberately moving the cell to edit mode on cell click, which doesn't make sense especially when the filter cell is empty.

I [tried](https://github.com/mui/mui-x/pull/7760/commits/45e6be0d5d8d8ac50bfcc6f2292c58e3b72cd919) to fix it, here's the updated codesandbox: https://codesandbox.io/s/aged-voice-jnvow4?file=/demo.tsx

Let me know your thoughts on it?

> Nit: Another option would be offloading keyboard navigation to header filter components.
If we allow them to control whether navigation keys should move the focus to a different cell or not, then we might not even need the global state for this.

You mean move remove the Header Filters' related logic from `useKeyboardNavigation` and `useGridFocus` and move focus and keyboard logic to GridHeaderFilterItem component?

Reply:
> You mean move remove the Header Filters' related logic from useKeyboardNavigation and useGridFocus and move it to GridHeaderFilterItem component?

I was thinking about being able to enable/disable keyboard navigation by the cell (e.g. when in edit mode).
But I don't think it's the way to go.
For the custom header filter components, they can use `startHeaderFilterEditMode`/`stopHeaderFilterEditMode` to leverage the editing state, right? If so, then we should be good 👍